### PR TITLE
Limit testing to PHPUnit <9.5

### PR DESCRIPTION
**Pull request by sanmai**
The root issue in PHPUnit-Bridge has been fixed already (https://github.com/symfony/symfony/issues/39329), but not in a release yet. 

We can remove this limitation once PHPUnit-Bridge >5.2.0 comes out.

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -288,16 +288,16 @@
         },
         {
             "name": "nikic/php-parser",
-            "version": "v4.10.2",
+            "version": "v4.10.3",
```
Comment (sidz) [inline composer.lock:291]: this change will break some our tests. I've already fixed our tests but didn't push code.

**Review by sidz — APPROVED** [review]

File: composer.lock (hunk 1 above)
Comment (sidz) [reply composer.lock:291]: fixed in https://github.com/infection/infection/pull/1454

Comment (sanmai) [inline composer.lock:291]:
```suggestion
            "version": "v4.10.2",
```
OK, let's try this.

File: composer.lock [hunk 2]
```diff
@@ -292,12 +292,12 @@
             "source": {
                 "type": "git",
                 "url": "https://github.com/nikic/PHP-Parser.git",
-                "reference": "658f1be311a230e0907f5dfe0213742aff0596de"
+                "reference": "dbe56d23de8fcb157bbc0cfb3ad7c7de0cfb0984"
```
Comment (sidz) [inline composer.lock:295]: reference should be changed as well.

**Review by maks-rafalko — APPROVED** [review]
fyi #1454 has been merge

**Pull request merged**